File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        Arrays.sort(participant);
        Arrays.sort(completion);
        
        int partNum = participant.length;

        for (int i = 0; i < completion.length; i++) {
            if (!participant[i].equals(completion[i])) {
                return participant[i]; 
            }
        }

        return participant[partNum-1];
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    answer = ''
    par = {}
    com = {}
    
    for p in participant:
        if p in par:
            par[p] += 1
        else:
            par[p] = 1
    
    for c in completion:
        par[c] -= 1
        
    for p in par:
        if par[p] != 0:
            answer = p
    
    return answer